frame/bridge/ethereum/issuing/src/lib.rs

```diff
@@ -0,0 +1,417 @@
+	{
+		/// register new erc20 token
+		RegisterErc20(AccountId, EthereumAddress),
+		/// redeem erc20 token
+		RedeemErc20(AccountId, EthereumAddress),
+		/// erc20 created
+		CreateErc20(EthereumAddress),
+		/// burn event
+		/// type: 1, backing, recipient, source, target, value
+		BurnToken(u8, EthereumAddress, EthereumAddress, EthereumAddress, EthereumAddress, U256),
+		/// token registered event
+		/// type: u8 = 0, backing, source(origin erc20), target(mapped erc20)
+		TokenRegistered(u8, EthereumAddress, EthereumAddress, EthereumAddress),
+	}
+}
+
+decl_storage! {
+	trait Store for Module<T: Config> as DarwiniaEthereumIssuing {
+		pub MappingFactoryAddress get(fn mapping_factory_address) config(): EthereumAddress;
+		pub EthereumBackingAddress get(fn ethereum_backing_address) config(): EthereumAddress;
+		pub VerifiedIssuingProof
+			get(fn verified_issuing_proof)
+			: map hasher(blake2_128_concat) EthereumTransactionIndex => bool = false;
+		pub BurnTokenEvents get(fn burn_token_events): Vec<<T as frame_system::Config>::Event>;
+	}
+}
+
+decl_module! {
+	pub struct Module<T: Config> for enum Call
+	where
+		origin: T::Origin
+	{
+		fn deposit_event() = default;
+
+		fn on_initialize(_n: BlockNumber<T>) -> Weight {
+			<BurnTokenEvents<T>>::kill();
+			0
+		}
+
+		#[weight = <T as darwinia_evm::Config>::GasWeightMapping::gas_to_weight(0x100000)]
+		pub fn register_erc20(origin, proof: EthereumReceiptProofThing<T>) {
+			log::debug!("start to register erc20 token");
+			let user = ensure_signed(origin)?;
+			let (tx_index, ethlog) = Self::verify_and_parse_proof(
+				Abi::register_event(),
+				proof)?;
+			let backing_address = EthereumBackingAddress::get();
+			let input = Self::abi_encode_token_creation(backing_address, ethlog)?;
+			Self::transact_mapping_factory(input)?;
+			VerifiedIssuingProof::insert(tx_index, true);
+			Self::deposit_event(RawEvent::RegisterErc20(user, backing_address));
+		}
+
+		#[weight = <T as darwinia_evm::Config>::GasWeightMapping::gas_to_weight(0x100000)]
+		pub fn redeem_erc20(origin, proof: EthereumReceiptProofThing<T>) {
+			log::debug!("start to redeem erc20 token");
+			let user = ensure_signed(origin)?;
+			let (tx_index, ethlog) = Self::verify_and_parse_proof(
+				Abi::backing_event(),
+				proof)?;
+			let backing_address = EthereumBackingAddress::get();
+			let input = Self::abi_encode_token_redeem(ethlog)?;
+			Self::transact_mapping_factory(input)?;
+			VerifiedIssuingProof::insert(tx_index, true);
+			Self::deposit_event(RawEvent::RedeemErc20(user, backing_address));
+		}
+	}
+}
+
+impl<T: Config> Module<T> {
+	/// The account ID of the issuing pot.
+	pub fn dvm_account_id() -> H160 {
+		let account32: AccountId32 = T::ModuleId::get().into_account();
+		let account20: &[u8] = &account32.as_ref();
+		H160::from_slice(&account20[..20])
+	}
+
+	/// get dvm ethereum unsigned transaction
+	pub fn unsigned_transaction(target: H160, input: Vec<u8>) -> dvm_ethereum::Transaction {
```

Comment:
```suggestion
	pub fn internal_transaction(target: H160, input: Vec<u8>) -> dvm_ethereum::Transaction {
```

Reply:
Even though this is an internal transaction, it will still be stored in ethereum block.

So we need to mock a transaction anyway.